### ISSUE TITLE
[java] GraphQL Connected Services

### ISSUE THREAD

**Issue by smeubank** [getsentry/sentry-java#2790]
### Background
We want to provide better GraphQL support with Sentry SDKs, frontend to backend. Work has already begun on some mobile platforms. There is also already some implementations for GraphQL server and client side. With the current work we want to ensure that the strategy to ensure proper connected errors and grouping works from backend to frontend. Java already has an [integration for GraphQL](https://docs.sentry.io/platforms/java/configuration/integrations/graphql/).

### Proposal

Java SDK should set context for GraphQL on errors same as what is done for Mobile SDK to ensure the value add from Frontend to Backend

Definition of Done TBC:

- [ ] GraphQL Context
- [ ] Propagating Trace

Further context:

https://github.com/getsentry/sentry/issues/50230
- https://github.com/getsentry/sentry/pull/50764

- Ideally sentry-graphql, we could do code sharing with sentry-apollo, both written in Java
    - Works on Spring https://spring.io/projects/spring-graphql

**Comment by marandaneto** [comment]
https://github.com/getsentry/sentry-java/tree/main/sentry-apollo-3 already supports the `captureFailedRequests` feature (GraphQL Client errors).

TLDR:

- Opt-in by default due to PII (requires request and response payload).
- Set `event.request.api_target` to `graphql`
- Populate the `event.request` including the `data` field (request payload)
- Populate the `event.contexts.response` including the `data` field (response payload)
- Set `event.fingerprints` to [`$operationName`, `$operationType`, `$statusCode`]
- Set `event.values.exception.mechanism.type` so we can track adoption.
- Propagating Trace (without performance)
